src/mintermization.cc

```diff
@@ -0,0 +1,297 @@
+            return OptionalBdd(OptionalBdd::NOTHING_E);
+        if (symbol_to_bddvar.count(node.name)) {
+            return OptionalBdd(symbol_to_bddvar.at(node.name));
+        } else {
+            BDD res = bdd_mng.bddVar();
+            symbol_to_bddvar[node.name] = res;
+            return OptionalBdd(res);
+        }
+    } else if (node.is_operator()) {
+        if (node.operator_type == FormulaNode::AND) {
+            assert(graph.children.size() == 2);
+            const OptionalBdd op1 = graph_to_bdd_afa(graph.children[0]);
+            const OptionalBdd op2 = graph_to_bdd_afa(graph.children[1]);
+            return op1 * op2;
+        } else if (node.operator_type == FormulaNode::OR) {
+            assert(graph.children.size() == 2);
+            const OptionalBdd op1 = graph_to_bdd_afa(graph.children[0]);
+            const OptionalBdd op2 = graph_to_bdd_afa(graph.children[1]);
+            return op1 + op2;
+        } else if (node.operator_type == FormulaNode::NEG) {
+            assert(graph.children.size() == 1);
+            const OptionalBdd op1 = graph_to_bdd_afa(graph.children[0]);
+            return !op1;
+        } else
+            assert(false && "Unknown type of operation. It should conjunction, disjunction, or negation.");
+    }
+
+    assert(false);
+}
+
+const BDD Mata::Mintermization::graph_to_bdd_nfa(const FormulaGraph &graph)
+{
+    const FormulaNode& node = graph.node;
+
+    if (node.is_operand()) {
+        if (symbol_to_bddvar.count(node.name)) {
+            return symbol_to_bddvar.at(node.name);
+        } else {
+            BDD res = bdd_mng.bddVar();
+            symbol_to_bddvar[node.name] = res;
+            return res;
+        }
+    } else if (node.is_operator()) {
+        if (node.operator_type == FormulaNode::AND) {
+            assert(graph.children.size() == 2);
+            const BDD op1 = graph_to_bdd_nfa(graph.children[0]);
+            const BDD op2 = graph_to_bdd_nfa(graph.children[1]);
+            return op1 * op2;
+        } else if (node.operator_type == FormulaNode::OR) {
+            assert(graph.children.size() == 2);
+            const BDD op1 = graph_to_bdd_nfa(graph.children[0]);
+            const BDD op2 = graph_to_bdd_nfa(graph.children[1]);
+            return op1 + op2;
+        } else if (node.operator_type == FormulaNode::NEG) {
+            assert(graph.children.size() == 1);
+            const BDD op1 = graph_to_bdd_nfa(graph.children[0]);
+            return !op1;
+        } else
+            assert(false);
+    }
+
+    assert(false);
+}
+
+void Mata::Mintermization::minterms_to_aut(Mata::IntermediateAut& res, const Mata::IntermediateAut& aut,
+                                           const std::vector<BDD>& minterms)
+{
+    for (const auto& trans : aut.transitions) {
+            // for each t=(q1,s,q2)
+        const auto &symbol_part = trans.second.children[0];
+
+        size_t symbol = 0;
+        assert(trans_to_bddvar.count(&symbol_part));
```

Comment:
I think this assert should now be a proper test (if count==0 then we should continue), `trans_to_bddvar` will now not map symbol parts of transitions that have bdd equal to 0.

Reply:
You're right I just fixed AFA version and forgot about this one.